src/isl/util.rs

```diff
@@ -27,6 +29,224 @@ pub enum Range {
 }
 
 impl Range {
+    /// Provides a boolean value to specify whether the given value is within the range or not
+    pub fn contains(&self, value: &OwnedElement) -> IonSchemaResult<bool> {
+        use RangeBoundaryValue::*;
+        use RangeBoundaryValueType::*;
+        match self {
+            Range::Integer(start, end) => {
+                match (start, end) {
+                    (Min, Max) => {
+                        unreachable!("Range boundaries can not be min and max together (i.e. range::[min, max] is not allowed)")
```

Comment:
Are all of these `unreachable!` states being checked when the `Range` is being constructed? For example, is this `unreachable!` because we proved that the range wasn't `range::[min, max]` when we parsed it?

I'm confirming that this is a code path we genuinely don't expect to encounter. If this is the first time we're checking for this case, we should return an error instead.

Reply:
Yes, its been checked while the construction of the range.